rpcq/_utils.py

```diff
@@ -41,33 +45,41 @@ def rpc_request(method_name: str, *args, **kwargs) -> rpcq.messages.RPCRequest:
     )
 
 
-def rpc_reply(id: Union[str, int], result: Optional[object]) -> rpcq.messages.RPCReply:
+def rpc_reply(id: Union[str, int], result: Optional[object],
+              warnings: List[Warning] = None) -> rpcq.messages.RPCReply:
     """
     Create RPC reply
 
     :param str|int id: Request ID
     :param result: Result
+    :param warning: List of warnings to attach to the message
     :return: JSON RPC formatted dict
     """
+    warnings = warnings or list()
```

Comment:
Why not have this be the default arg instead of None?

Reply:
If you have a mutable default argument in Python the reference gets bound at parse time and then every call to the function gets the same reference.

Reply:
good call (also good foot gun)